GDJS/Runtime/howler-sound-manager/howler-sound-manager.js

```diff
@@ -14,87 +15,274 @@
  * @memberof gdjs
  * @class HowlerSound
  */
-gdjs.HowlerSound = function (o) {
-  Howl.call(this, o);
-  this._paused = false;
-  this._stopped = true;
-  this._canBeDestroyed = false;
-  this._rate = o.rate || 1;
-
-  //Add custom events listener to keep
-  //track of the sound status.
-  var that = this;
-  this.on('end', function () {
-    if (!that.loop()) {
-      that._canBeDestroyed = true;
-      that._paused = false;
-      that._stopped = true;
+gdjs.HowlerSound = class HowlerSound {
+  /**
+   * The ID of the played sound.
+   * @type {?number}
+   * @private
+   */
+  _id = null;
+
+  /**
+   * The Howl passed to the constructor.
+   * It defines the sound file that is being played.
+   */
+  _howl = null;
+
+  /**
+   * Internal variable to check if the sound is still loading.
+   * A loading sound will count as played.
+   * @type {boolean}
+   * @private
+   */
+  _loading = false;
+
+  /**
+   * An internal variable for tracking if the sound can be destroyed.
+   * This is used to remove a free sound when it finishes.
+   * @type {boolean}
+   * @private
+   */
+  _canBeDestroyed = false;
+
+  constructor(howl) {
+    this._howl = howl;
+  }
+
+  /**
+   * Begins playback of the sound.
+   * @returns {HowlerSound} The current instance for chaining.
+   */
+  play() {
+    if (this._howl._state === 'loaded') {
```

Comment:
Seems that you can use `state()`: https://github.com/goldfire/howler.js/blob/97b10106c244e0f95077561aae1d512522c6d015/src/howler.core.js#L1722